gufe/storage/pseudodirectory.py

```diff
@@ -0,0 +1,156 @@
+from typing import Union, Optional
+from pathlib import Path
+from os import PathLike
+from .externalresource import ExternalStorage
+
+import logging
+_logger = logging.getLogger(__name__)
+
+class SharedRoot:
```

Comment:
comments, it's not obvious what a `SharedRoot` is etc

Reply:
I'm up for name changes if there's a suggestion. I don't particularly like these names -- I originally was calling `SharedRoot` a `PseudoDirectory`, but I thought that would be more confusing.

Reply:
I don't mind what it's called, but it'll need a one liner to explain what the class is/does either way